app/src/main/java/org/opencovidtrace/octrace/MainActivity.kt

```diff
@@ -87,6 +103,23 @@ class MainActivity : AppCompatActivity() {
 
             startActivity(intent)
         }
+        handleDeepLink()
+    }
+
+    private fun handleDeepLink() {
+        val data: Uri? = intent.data
+        if (data != null && data.isHierarchical) {
+            val uri = Uri.parse(intent.dataString)
+            val id = uri.getQueryParameter("i")
+            val key = uri.getQueryParameter("k")
+            val token = uri.getQueryParameter("d")
+            val platform = uri.getQueryParameter("p")
+            val tst = uri.getQueryParameter("t")?.toLongOrNull()
+            ifAllNotNull(id, key, token, platform, tst) { i, k, d, p, t ->
```

Comment:
Plz use normal names as above. Params are single letter because for QR content size is critical.